src/components/KDropdown/KDropdownItem.vue

```diff
@@ -1,12 +1,12 @@
 <template>
   <li
     class="k-dropdown-item"
-    :class="{
+    :class="[{
       'has-divider': hasDivider,
       'disabled': disabled,
       'danger': danger || isDangerous,
-      'dropdown-selected-option': selected
-    }"
+      'dropdown-selected-option': selected,
+    }, $attrs.class]"
```

Comment:
I'm confused, wasn't the point to bind the `data-testid` to the element that actually triggers the click (ie. the `button`, `li`, etc)? If the child `<component>` still has a `data-testid` (which it does - `dropdown-item-trigger`) won't we still have the original problem attempting to catch the clicks in DD?

Reply:
No - if we provide `data-testid` it will override the original value

Reply:
But how? I'm not understanding what this `availableComponent` thing is

Reply:
It binds provided attributes after it sanitizes them a bit: https://github.com/Kong/kongponents/blob/75576a48ae122e9be72d891638d00b925289f3cf/src/components/KDropdown/KDropdownItem.vue#L156